algorithms/HPdclassifier/R/predict.hpdegbm.R

```diff
@@ -0,0 +1,218 @@
+     npartition_test <- npartitions(newdata)  
+     nExecutor_test  <- npartition_test
+
+     nTest <- nrow(newdata)
+     daPredict <- darray(dim=c(nTest,1), blocks=c(ceiling(nTest/npartition_test),1), sparse=FALSE)
+
+
+     # distributed prediction
+     foreach(i, 1:nExecutor_test, function(GBM_model=model, best.iter=best.iter, predi=splits(daPredict,i), data2=splits(newdata,i),appType=appType, type=type) { 
+        library(gbm)
+
+        # Gaussian distribution for regression
+        if (appType == "regression") {
+           # Prediction by the first sub-model: AdaBoost, Bernoulli distribution for binary classification
+           GBM_modeli <- GBM_model[[1]]
+           best.iteri <- best.iter[1,1]
+           predi <- predict(GBM_modeli, newdata, best.iteri, type="link")  
+
+           # fusion with other sub-models
+           npartition_train <- nrow(best.iter)
+           for (k in 2: npartition_train) {
+               GBM_modelk <- GBM_model[[k]]
+               best.iterk <- best.iter[k,1]
+               predi <- predi + predict(GBM_modelk, newdata, best.iterk, type="link")
+           } 
+           predi <- predi/npartition_train # average of regressions from sub-models
+         }
+
+         # bernoulli or AdaBoost distribution for binary classification
+         if (appType == "binary-classification") {
+           # Prediction by the first sub-model: AdaBoost, Bernoulli distribution for binary classification
+           GBM_modeli <- GBM_model[[1]]
+           best.iteri <- best.iter[1,1]
+           predi < - sign(predict(GBM_modeli, data2, best.iteri, type)) # AdaBoost/Bernoulli 
+
+           # fusion with other sub-models
+           npartition_train <- nrow(best.iter)
+           for (k in 2: npartition_train) {
+               GBM_modelk <- GBM_model[[k]]
+               best.iterk <- best.iter[k,1]
+               predi <- predi + sign(predict(GBM_modelk, data2, best.iterk, type))
+           } 
+           predi <- sign(predi)
+         }
+
+
+         # multinomial distribution for multi-class classification
+         if (appType == "multi-classification") {
+           # Prediction by the first sub-model
+           # X: dframe/darray; Y: dframe (factor vector originally)/darray
+           GBM_modeli <- GBM_model[[1]]
+           best.iteri <- best.iter[1,1]
+           pred0 <- predict(GBM_modeli, data2, best.iteri, type="response") # fuzzy classification of multi-class
+
+           # fusion with other sub-models
+           npartition_train <- nrow(best.iter)
+           for (k in 2: npartition_train) {
+               GBM_modelk <- GBM_model[[k]]
+               best.iterk <- best.iter[k,1]
+               pred0 <- pred0 + (predict(GBM_modelk, data2, best.iterk, type="response")) # fusion of sub-models
+           } 
+           predi <- apply(pred0,1,which.max)
+         }
+
+        update(predi)
+     }) 
+
+     Predictions <- getpartition(daPredict)
```

Comment:
You previously assumed that dframe,darray correspond to big data so you shouldn't getpartition the predictions because master is supposed to be lightweight.